demos/https/iot_demo_https_common.c

```diff
@@ -0,0 +1,227 @@
+     * we are ignoring. */
+    httpsClientStatus = IotHttpsClient_AddHeader( fileSizeReqHandle, RANGE_HEADER_FIELD, RANGE_HEADER_FIELD_LENGTH, RANGE_0_TO_0_HEADER_VALUE, RANGE_0_TO_0_HEADER_VALUE_LENGTH );
+
+    if( httpsClientStatus != IOT_HTTPS_OK )
+    {
+        IotLogError( "Failed to write the header \"Range: bytes=0-0\" into the request. With error code: %d", httpsClientStatus );
+        return EXIT_FAILURE;
+    }
+
+    /* Send the request synchronously. */
+    httpsClientStatus = IotHttpsClient_SendSync( connHandle, fileSizeReqHandle, &fileSizeRespHandle, &fileSizeRespConfig, 0 );
+
+    if( httpsClientStatus != IOT_HTTPS_OK )
+    {
+        IotLogError( "There has been an error receiving the response. The error code is: %d", httpsClientStatus );
+
+        /* Even though there is an error we could still read headers and body that may have been
+         * filled in the buffer depending on the error. */
+    }
+
+    /* Read the response status to see if we successfully got a partial content response. If we did not then
+     * we failed and will exit the demo. */
+    httpsClientStatus = IotHttpsClient_ReadResponseStatus( fileSizeRespHandle, &respStatus );
+
+    if( httpsClientStatus != IOT_HTTPS_OK )
+    {
+        IotLogError( "Could not read the response status. Error code %d", httpsClientStatus );
+        return EXIT_FAILURE;
+    }
+
+    if( respStatus != IOT_HTTPS_STATUS_PARTIAL_CONTENT )
+    {
+        IotLogError( "Could not retrieve the file size. s3 responded with response status: %d", respStatus );
+
+        if( respStatus == IOT_HTTPS_STATUS_TEMPORARY_REDIRECT )
+        {
+            IotLogInfo( "If the bucket was created less than 24 hours ago, you must add the region the bucket was created to the presigned URL. " );
+            IotLogInfo( "You can change the Origin Domain Name from bucketname.s3.amazonaws.com to bucketname.s3-<region>.amazonaws.com." );
+        }
+
+        if( respStatus == IOT_HTTPS_STATUS_FORBIDDEN )
+        {
+            IotLogInfo( "Pre-signed URL may have expired or you do not have permissions on this account to access the S3 object." );
+        }
+
+        if( respStatus == IOT_HTTPS_STATUS_NOT_FOUND )
+        {
+            IotLogInfo( "The specific object key in the presigned URL was not found in the bucket. " );
+        }
+
+        return EXIT_FAILURE;
+    }
+
+    /* Get the file size by parsing the "bytes 0-0/FILESIZE" Content-Range header value string. */
+    httpsClientStatus = IotHttpsClient_ReadHeader( fileSizeRespHandle, CONTENT_RANGE_HEADER_FIELD, CONTENT_RANGE_HEADER_FIELD_LENGTH, contentRangeValStr, sizeof( contentRangeValStr ) );
+
+    if( httpsClientStatus != IOT_HTTPS_OK )
+    {
+        IotLogError( "Could find the Content-Range header in the response. Error code %d", httpsClientStatus );
+        return EXIT_FAILURE;
+    }
+
+    pFileSizeStr = strstr( contentRangeValStr, "/" ) + sizeOfOneChar;
```

Comment:
What if the server didn't return a '/', or a '\0', before we read off the end of contentRangeValStr?

Reply:
Good point. I will need to check for NULL returned.